### PR TITLE
Implement Insert

### PR DESCRIPTION
**Pull request by sumeetgajjar**

### CODE REVIEW

File: crdt_lib/include/types.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//
+// Created by sumeet on 11/22/20.
+//
+
+#ifndef NUPAD_TYPES_H
+#define NUPAD_TYPES_H
+
+#include <string>
+#include <unordered_map>
+#include <sstream>
```
Comment (madhurjain2095) [inline crdt_lib/include/types.h]: unused import

File: crdt_lib/include/list.h [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+#include "operation.h"
+
+namespace nupad::crdt {
+
+    /* All definitions are in-place to avoid linker error, this is a standard practice
+     * used in STL.
+     * ref: https://stackoverflow.com/questions/8752837/undefined-reference-to-template-class-constructor
+     */
+    template<typename T>
+    class CRDTList {
```
Comment (madhurjain2095) [inline crdt_lib/include/list.h]: @sumeetgajjar  says: Rename to `DoublyLinkedList`

**Review by madhurjain2095 — APPROVED** [review]

File: crdt_lib/include/types.h (hunk 1 above)
Comment (sumeetgajjar) [reply crdt_lib/include/types.h]: Done

File: crdt_lib/include/list.h (hunk 2 above)
Comment (sumeetgajjar) [reply crdt_lib/include/list.h]: Done

**Pull request merged**